options/options.go

```diff
@@ -122,20 +129,29 @@ func InterpretOptionsLenient(linked linker.Result, opts ...InterpreterOption) (s
 // effectively ignored. Any options that are uninterpretable (due to such
 // errors) will remain in the "uninterpreted_option" fields.
 func InterpretUnlinkedOptions(parsed parser.Result, opts ...InterpreterOption) (sourceinfo.OptionIndex, error) {
-	return interpretOptions(true, noResolveFile{parsed}, nil, reporter.NewHandler(nil), opts)
+	return interpretOptions(true, noResolveFile{parsed}, nil, nil, opts)
 }
 
 func interpretOptions(lenient bool, file file, res linker.Resolver, handler *reporter.Handler, interpOpts []InterpreterOption) (sourceinfo.OptionIndex, error) {
-	interp := interpreter{
+	interp := &interpreter{
 		file:       file,
 		resolver:   res,
 		lenient:    lenient,
-		reporter:   handler,
 		index:      sourceinfo.OptionIndex{},
 		pathBuffer: make([]int32, 0, 16),
 	}
+	var errHandler internal.ErrorHandler
+	switch {
+	case lenient:
+		errHandler = (*lenientHandler)(interp)
+	case handler == nil:
+		errHandler = reporter.NewHandler(nil)
```

Comment:
Could the nil case avoid an allocation here by `*reporter.Handler` checking if itself is nil?

Reply:
The previous formulation never bothered with this, so Perhaps I should just remove it and leave it be a separate nil hazard. In practice, nothing ever calls the non-lenient entry point with a nil handler.